Python/Source/Lift.cpp

```diff
@@ -0,0 +1,90 @@
+#include <vector>
+#include "pybind11/functional.h"
+#include "Autobox.hpp"
+#include "Rover/Box.hpp"
+#include "Rover/Evaluator.hpp"
+
+using namespace pybind11;
+
+namespace {
+  using namespace Rover;
+  using ArgsCont = std::vector<Box<object>>;
+  using KwargsCont = std::vector<std::pair<handle,
+    Box<object>>>;
+  
+  ArgsCont args_apply(const std::function<Box<object>(object&&)>& func,
+      tuple&& args) {
+    auto result = ArgsCont();
+    for(auto i = 0U; i < args.size(); ++i) {
```

Comment:
It didn't, because tuple_iterators returned by .begin() and .end() (that range-based for loop rely on) are in fact constant iterators (they reference const handles). operator[] though references (non-const) objects as required.

Reply:
Okay good catch, that's absolutely a defect in pybind11 but we have to live with it.